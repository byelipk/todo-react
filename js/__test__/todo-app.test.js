import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import TodoApp from '../todo-app';
import TodoItem from '../todo-item';

test('Jest works', () => {
  expect(1).toEqual(1);
});

test('It renders correctly', () => {
  const app = shallow( <TodoApp /> );

  expect(app).toMatchSnapshot();
});

test('It renders within the correct container', () => {
  const app = shallow( <TodoApp /> );

  expect(app.find('.container').length).toEqual(1);
});

test('It renders the correct title', () => {
  const app = shallow( <TodoApp title='Sherlock' /> );

  expect(app.find('h1').length).toEqual(1);
  expect(app.find('h1').text()).toEqual('Sherlock');
});

test('It adds item to list', () => {
  const app = mount( <TodoApp /> );

  const input = app.find('input[type="text"]');
  const form  = app.find('form');

  expect(app.find('li').length).toEqual(2);

  input.simulate('change', { target: { value: 'Do more work' } });
  form.simulate('submit');

  expect(app.find('li').length).toEqual(3);
});

test('It removes item from list', () => {
  const app = mount( <TodoApp todos={['task1', 'task2']} /> );

  expect(app.find('ul').children().length).toEqual(2);

  const todo = app.find('ul').childAt(0);
  const deleteButton = todo.find('button');
  deleteButton.simulate('click');

  expect(app.find('ul').children().length).toEqual(1);
});
