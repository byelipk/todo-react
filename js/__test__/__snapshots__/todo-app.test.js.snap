// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders correctly 1`] = `
<div
  className="container"
>
  <h1>
    Todo List
  </h1>
  <TodoList
    deleteEntry={[Function]}
    todos={
      Array [
        "get food",
        "drink water",
      ]
    }
  />
  <TodoForm
    createEntry={[Function]}
  />
</div>
`;
